internal/conf/rate.go

```diff
@@ -0,0 +1,93 @@
+package conf
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/didip/tollbooth/v5"
+	"github.com/didip/tollbooth/v5/limiter"
+)
+
+type Rate struct {
+	Events   float64       `json:"events,omitempty"`
+	OverTime time.Duration `json:"over_time,omitempty"`
+}
+
+func (r *Rate) EventsPerSecond() float64 {
+	if int64(r.OverTime) == 0 {
+		return r.Events
+	}
+
+	return r.Events / r.OverTime.Seconds()
+}
+
+// DivideAndSetDuration is to only be used when the rate is in the legacy format, i.e. only specifies Events without specifying the duration.
+func (r *Rate) DivideAndSetDuration(div float64, d time.Duration) Rate {
+	if r.OverTime == time.Duration(0) {
+		return Rate{
+			Events:   r.Events / div,
+			OverTime: d,
+		}
+	}
+
+	return *r
+}
+
+func (r *Rate) DivideIfDefaultDuration(div float64) *Rate {
+	if r.OverTime == time.Duration(0) {
+		return &Rate{
+			Events: r.Events / div,
+		}
+	}
+
+	return r
+}
+
+func (r *Rate) CreateLimiter() *limiter.Limiter {
+	overTime := r.OverTime
+	if int64(overTime) == 0 {
+		// if r.OverTime is not specified, i.e. the configuration specified just a single float64 number, the
+		overTime = time.Hour
+	}
+
+	return tollbooth.NewLimiter(r.EventsPerSecond(), &limiter.ExpirableOptions{
+		DefaultExpirationTTL: overTime,
+	})
+}
+
+func (r *Rate) Decode(value string) error {
```

Comment:
can we add a comment to describe that decode is necessary for envconfig to unmarshal the value correctly?